src/Symfony/Component/Notifier/Bridge/MicrosoftTeams/MicrosoftTeamsOptions.php

```diff
@@ -0,0 +1,147 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Notifier\Bridge\MicrosoftTeams;
+
+use Symfony\Component\Notifier\Bridge\MicrosoftTeams\Action\ActionInterface;
+use Symfony\Component\Notifier\Bridge\MicrosoftTeams\Section\Section;
+use Symfony\Component\Notifier\Bridge\MicrosoftTeams\Section\SectionInterface;
+use Symfony\Component\Notifier\Exception\InvalidArgumentException;
+use Symfony\Component\Notifier\Message\MessageOptionsInterface;
+use Symfony\Component\Notifier\Notification\Notification;
+
+/**
+ * @author Edouard Lescot <edouard.lescot@gmail.com>
+ * @author Oskar Stark <oskarstark@googlemail.com>
+ *
+ * @see https://docs.microsoft.com/en-us/outlook/actionable-messages/message-card-reference
+ */
+final class MicrosoftTeamsOptions implements MessageOptionsInterface
+{
+    private const MAX_ACTIONS = 4;
+
+    private $options = [];
+
+    public function __construct(array $options = [])
```

Comment:
MessageCard default settings will avoid any issues (without these options settled down explicitely, MessageCard will show only title & text skipping the rest of options). On the other hand, this setup does not impact at all the simple text messaging.

```php
  /**
     * MicrosoftTeamsOptions constructor.
     * @param array|string[] $options
     */
    public function __construct(array $options = [
        "MessageCard",
        "https://schema.org/extensions"
    ])
```

Reply:
I don't understand what you mean?

Is it only working when you provide context and type via the constructor? 🤔

Reply:
I have rechecked one again, when `"MessageCard",
  "https://schema.org/extensions"` are setup (or not) in __construct the message in Teams is sent as expected - which is great. Simply the structure of JSON does not fit exactly the Microsoft documentation. I would set those two by default avoiding any issues. see: 
https://docs.microsoft.com/fr-fr/microsoftteams/platform/webhooks-and-connectors/how-to/connectors-using

Reply:
In addition, the image part of section seems to be not working properly - see below example:

```
  (new Section())
      ->image((new Image())
          ->image('https://symfony.com/logos/symfony_black_03.png')
          ->title('The Symfony Logo')
      )
```

I have tested with several images and different sources - same output.

This needs to be invesigated.

![image](https://user-images.githubusercontent.com/13560497/116003543-abf26d80-a5fe-11eb-8675-32fee373d20c.png)

Reply:
This looks like a problem on the MS side, lets ignore that

Reply:
> I have rechecked one again, when "MessageCard", "https://schema.org/extensions" are setup (or not) in __construct the message in Teams is sent as expected - which is great. Simply the structure of JSON does not fit exactly the Microsoft documentation. I would set those two by default avoiding any issues. see:
https://docs.microsoft.com/fr-fr/microsoftteams/platform/webhooks-and-connectors/how-to/connectors-using

So it is working, but the JSON looks different?
Can you show the real diff?

Reply:
Simply "@type" & "@context" are missing in the beginning of JSON, that is all.

```json
   "@type": "MessageCard",
   "@context": "http://schema.org/extensions",
...
```

Reply:
But why, we call `toArrray` in the Transport and:
```php
    public function toArray(): array
      {
          $options = $this->options;
 
          // Send a text, not a message card
          if (1 === \count($options) && isset($options['text'])) {
              return $options;
          }
 
          $options['@type'] = 'MessageCard';
          $options['@context'] = 'https://schema.org/extensions';
 
          return $options;
      }
```

the method adds `@context` and `@type` every time 🤔

Reply:
You are right Oskar!, it works as expected, I had an internal error linked to code modified in the paskage. Please ignore.

Reply:
Glad to hear that it works 👍